onnxruntime/core/providers/shared/node_unit/node_unit.cc

```diff
@@ -79,13 +81,77 @@ bool IsVariadicQLinearOp(QLinearOpType type) {
   return type == QLinearOpType::QLinearConcat;
 }
 
+const std::vector<const Node*> GetQDQOutputNodes(const GraphViewer& graph_viewer, const QDQ::NodeGroup& node_group) {
+  std::vector<const Node*> output_nodes;
+  output_nodes.reserve(node_group.q_nodes.size());
+  for (const auto& node_idx : node_group.q_nodes) {
+    output_nodes.push_back(graph_viewer.GetNode(node_idx));
+  }
+  return output_nodes;
+}
+
+// Get the input or output NodeUnitIODef(s) for the given QDQ NodeGroup
+void GetQDQIODefs(const Node& target_node, const QDQ::NodeGroup& node_group,
+                  std::vector<NodeUnitIODef>& io_defs, bool is_input) {
+  const auto& dq_or_q_nodes = is_input ? node_group.dq_nodes : node_group.q_nodes;
+  const auto target_node_io_defs = is_input ? target_node.InputDefs() : target_node.OutputDefs();
+  const size_t target_node_io_defs_size = target_node_io_defs.size();
+
+  // Find all the quantized IO defs and indices (for the input to the target node)
+  std::unordered_map<size_t, NodeUnitIODef> quantized_io_defs;
+  quantized_io_defs.reserve(target_node_io_defs_size);
+
+  auto cur = is_input ? target_node.InputEdgesBegin() : target_node.OutputEdgesBegin();
+  auto end = is_input ? target_node.InputEdgesEnd() : target_node.OutputEdgesEnd();
+  for (; cur != end; ++cur) {
+    const Node& node = cur->GetNode();
+
+    // If we can find the node index in the dq or q nodes, then this is a quantize node (can be DQ or Q depends on is_input)
+    if (std::find(dq_or_q_nodes.cbegin(), dq_or_q_nodes.cend(), node.Index()) != dq_or_q_nodes.cend()) {
+      auto idx = cur->GetDstArgIndex();
+      const auto node_inputs = node.InputDefs();
```

Comment:
Does this need to take into account if it's input or output? Would have expected we want the dst arg index for inputs to the DQ nodes and the src arg index for outputs from the Q nodes.